### PR TITLE
Added the report cloudwatch metrics to send custom metrics to AWS cloudwatch service

### PR DESCRIPTION
**Pull request by rakhimundhada15**
### Problem Description

Ex. In s3 collector , need to check how many time file not found issue is occur for particular customer.

### Solution Description
Added the report cloudwatch metrics to send custom metrics to AWS cloudwatch service. Use this to create the custom metrics in all AWS collector

### CODE REVIEW

File: statistics_templates.js [hunk 1]
```diff
@@ -157,6 +157,43 @@ var getAllKinesisMetricsFuns = function(streamName) {
...
+ * @param {*} metricName - custom metric name 
+ * @param {*} collectorType - Collector type 
+ * @param {*} functionName - Lambda function name
+ * @param {*} namespace - Custom namespace 
+ * @param {*} StandardUnit - can be any value from this or string "Seconds"|"Microseconds"|"Milliseconds"|"Bytes"|"Kilobytes"|"Megabytes"|"Gigabytes"|"Terabytes"|"Bits"|"Kilobits"|"Megabits"|"Gigabits"|"Terabits"|"Percent"|"Count"|"Bytes/Second"|"Kilobytes/Second"|"Megabytes/Second"|"Gigabytes/Second"|"Terabytes/Second"|"Bits/Second"|"Kilobits/Second"|"Megabits/Second"|"Gigabits/Second"|"Terabits/Second"|"Count/Second"|"None"|string;
+ * @param {*} unitValue -value as per StandardUnit selected
+ * @param {*} callback 
+ * @returns 
+ */
+var reportCWMetric = function (metricName, collectorType, functionName, namespace, standardUnit, unitValue, callback) {
```
Comment (kkuzmin) [inline statistics_templates.js:171]: please move this function to [`al_aws.js`](https://github.com/alertlogic/al-aws-collector-js/blob/master/al_aws.js)

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: statistics_templates.js (hunk 1 above)
Comment (rakhimundhada15) [reply statistics_templates.js:171]: ok

File: al_aws.js [hunk 2]
```diff
@@ -141,6 +141,43 @@ var getKinesisMetrics = function (streamName, metricName, statistics, callback)
...
+ * @param {*} metricName - custom metric name 
+ * @param {*} collectorType - Collector type 
+ * @param {*} functionName - Lambda function name
+ * @param {*} namespace - Custom namespace 
+ * @param {*} StandardUnit - can be any value from this or string "Seconds"|"Microseconds"|"Milliseconds"|"Bytes"|"Kilobytes"|"Megabytes"|"Gigabytes"|"Terabytes"|"Bits"|"Kilobits"|"Megabits"|"Gigabits"|"Terabits"|"Percent"|"Count"|"Bytes/Second"|"Kilobytes/Second"|"Megabytes/Second"|"Gigabytes/Second"|"Terabytes/Second"|"Bits/Second"|"Kilobits/Second"|"Megabits/Second"|"Gigabits/Second"|"Terabits/Second"|"Count/Second"|"None"|string;
+ * @param {*} unitValue -value as per StandardUnit selected
+ * @param {*} callback 
+ * @returns 
+ */
+ var reportCWMetric = function (metricName, collectorType, functionName, namespace, standardUnit, unitValue, callback) {
```
Comment (kkuzmin) [inline al_aws.js:155]: actually let's have this as a part of `AlAwsCollector` class

**Review by kkuzmin — CHANGES_REQUESTED** [review]

File: al_aws_collector.js [hunk 3]
```diff
@@ -924,6 +924,44 @@ class AlAwsCollector {
...
+ * metricName :'custom metrics'
+ * nameSpace : 'PAWSCollector',
+ * standardUnit: 'Count',
+ * unitValue :1
+ * }
+ * @param {*} callback 
+ * @returns 
+ */
+    reportCWMetric(param, callback) {
+        let cloudwatch = new AWS.CloudWatch({ apiVersion: '2010-08-01' });
```
Comment (kkuzmin) [inline al_aws_collector.js:941]: let's init `cloudwatch` object only once in constructor and then use it here.

**Review by kkuzmin — CHANGES_REQUESTED** [review]

**Review by kkuzmin — APPROVED** [review]

**Pull request merged**